pkg/kubelet/stats/cadvisor_stats_provider.go

```diff
@@ -134,15 +134,65 @@ func (p *cadvisorStatsProvider) ListPodStats() ([]statsapi.PodStats, error) {
 	for _, podStats := range podToStats {
 		// Lookup the volume stats for each pod.
 		podUID := types.UID(podStats.PodRef.UID)
+		var ephemeralStats []statsapi.VolumeStats
 		if vstats, found := p.resourceAnalyzer.GetPodVolumeStats(podUID); found {
-			podStats.VolumeStats = vstats.Volumes
+			podStats.VolumeStats = append(vstats.EphemeralVolumes, vstats.PersistentVolumes...)
```

Comment:
This might append `vstats.PersistentVolumes` to `vstats.EphemeralVolumes`. To avoid that you need to create a new slice.

Reply:
fixed